src/vs/vscode.d.ts

```diff
@@ -4452,6 +4452,11 @@ declare module 'vscode' {
 		 * Args for the custom shell executable, this does not work on Windows (see #8429)
 		 */
 		shellArgs?: string[];
+		/**
+		 * A custom environment for the terminal, if this is not set the environment will be inherited
```

Comment:
Asked people's thoughts in https://github.com/Microsoft/vscode/issues/20446#issuecomment-314267087